x-pack/docs/en/security/securing-communications/update-tls-certificates.asciidoc

```diff
@@ -0,0 +1,784 @@
+====
+   `-in`:: Specifies the name of the input YAML file (`instances.yml`)
+   containing information about each of your nodes.
+
+   `--ca <ca_file>`:: Name of the CA keystore used to sign your certificates.
+====
+
+   a. Enter a name for the output file or accept the default of
+   `certificate-bundle.zip`.
+
+   b. When prompted, enter a password for each node certificate. It's good
+   practice to use a different password for each node certificate and store the
+   files in a secure location.
+
+   c. Decompress the `certificate-bundle.zip` file. The output contains a
+   directory for each of your nodes with a `<node-name>.p12` file.
+
+   d. Rename the `<node-name>.p12` file to match the existing file name
+   specified by the `xpack.security.transport.ssl.keystore.path` variable in 
+   the `elasticsearch.yml` file on each node.
+
+. If you entered a password when creating the node keystore that is different 
+from the current keystore password, run the following command to store the
+password in the {es} keystore:
++
+--
+[source,shell]
+----
+./bin/elasticsearch-keystore add xpack.security.transport.ssl.keystore.secure_password
+----
+--
+
+. [[start-rolling-restart,step 5]]On the current node in your cluster where you're updating the keystore,
+start a <<restart-cluster-rolling,rolling restart>>.
++
+Stop at the step indicating *Perform any needed changes*, and then proceed to
+the next step in this procedure.
+
+. [[replace-keystores]]Replace your existing keystore with the new keystore, ensuring that the
+file names match. For example, `node1.p12`.
++
+If your 
+<<cert-password-updates,keystore password is changing>>, then save the
+keystore with a new filename so that {es} doesn't attempt to reload the file
+before you update the password.
+
+. If you needed to save the new keystore with a new filename, update the
+`ES_PATH_CONF/elasticsearch.yml` file to use the filename of the new keystore.
+For example:
++
+[source,yaml]
+----
+xpack.security.transport.ssl.keystore.path: config/node1.p12
+xpack.security.transport.ssl.keystore.type: PKCS12
+xpack.security.transport.ssl.truststore.path: config/elastic-stack-ca.p12
+xpack.security.transport.ssl.truststore.type: PKCS12
+----
+
+. Start the node where you updated the keystore, and then complete the remaining
+steps for a <<restart-cluster-rolling,rolling restart>>.
+
+. [[verify-keystore,step 9]](Optional) Use the <<security-api-ssl,SSL certificate API>> to verify that
+{es} loaded the new keystore.
+
+. One node at a time, complete <<start-rolling-restart>> through
+<<verify-keystore>> until you've updated all keystores in your cluster.
+
+[discrete]
+[[transport-layer-sameca-whatsnext]]
+==== What's next?
+Well done! You've updated the keystore for the transport layer. You can also
+<<node-certs-same-http,update the keystore for the HTTP layer>> if
+necessary. If you're not updating the keystore for the HTTP layer, then you're
+all set. 
+
+[[node-certs-same-http]]
```

Comment:
splitting this like that makes it easy and readable, but it introduces the misconception that one _MUST DO_ 2 rolling restarts, one for transport and one for http, while in practice they could combine the rotation of both layer certificates in the same rolling restart. Can we add something related to taht to make it obvious or do you think it will just confuse users?

Reply:
The reason I split them is because users might be updating certificates only for the transport layer or only the HTTP layer. We could clarify the last step to say:

> . If you're only updating certificates for the transport layer (and not the HTTPY layer), then complete <<start-rolling-restart>> through <<verify-keystore>> one node at a time until you've updated all keystores in your cluster.
> 
> Otherwise, do not complete a rolling restart and complete the steps for generating a new certificate for the HTTP layer.